packages/host/app/components/operator-mode/code-submode/boxel-spec-preview.gts

```diff
@@ -0,0 +1,274 @@
+  Boxel Specification
+
+  <span class='has-boxel-spec' data-test-has-boxel-spec>
+    {{#if @showCreateBoxelSpecIntent}}
+      <Pill style={{htmlSafeColor 'orange'}}>No Boxel Spec</Pill>
+    {{/if}}
+  </span>
+
+  <style scoped>
+    .has-boxel-spec {
+      margin-left: auto;
+      color: var(--boxel-450);
+      font: 500 var(--boxel-font-xs);
+      letter-spacing: var(--boxel-lsp-xl);
+      text-transform: uppercase;
+    }
+  </style>
+</template>;
+
+interface ContentSignature {
+  Element: HTMLDivElement;
+  Args: {
+    showCreateBoxelSpecIntent: boolean;
+    boxelSpecInstances: CatalogEntry[];
+    selectedInstance: CatalogEntry | null;
+    createBoxelSpec: () => void;
+    selectBoxelSpec: (boxelSpec: CatalogEntry) => void;
+  };
+}
+
+const BoxelSpecPreviewContent: TemplateOnlyComponent<ContentSignature> =
+  <template>
+    <div class='boxel-spec-preview'>
+      {{#if @showCreateBoxelSpecIntent}}
+        <div class='create-boxel-spec-intent-message'>
+          Create a Boxel Specification to be able to create new instances
+        </div>
+        <BoxelButton @kind='primary' {{on 'click' @createBoxelSpec}}>
+          Create Boxel Spec
+        </BoxelButton>
+      {{else}}
+        <div class='boxel-spec-selector'>
+          <RadioInput
+            @groupDescription='Select Boxel Spec'
+            @items={{@boxelSpecInstances}}
+            @checkedId={{@selectedInstance.id}}
+            @orientation='vertical'
+            @spacing='compact'
+            as |item|
+          >
+            <item.component @onChange={{(fn @selectBoxelSpec item.data)}}>
+              {{item.data.id}}
+            </item.component>
+          </RadioInput>
+        </div>
+        {{#if @selectedInstance}}
+          {{#let (getComponent @selectedInstance) as |CardComponent|}}
+            <CardComponent />
+          {{/let}}
+        {{/if}}
+      {{/if}}
+    </div>
+
+    <style scoped>
+      .create-boxel-spec-intent-message {
+        text-align: center;
+        color: var(--boxel-450);
+        font-weight: 500;
+      }
+      .boxel-spec-preview {
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        flex-direction: column;
+        gap: var(--boxel-sp-sm);
+        padding-top: var(--boxel-sp);
+      }
+      .boxel-spec-selector {
+        padding: var(--boxel-sp-sm);
+      }
+    </style>
+  </template>;
+
+export default class BoxelSpecPreview extends GlimmerComponent<Signature> {
+  @service private declare operatorModeStateService: OperatorModeStateService;
+  @service private declare realm: RealmService;
+  @tracked selectedInstance: CatalogEntry | null = this.boxelSpecInstances[0];
+
+  get realmURL() {
+    return this.realm.realmOfURL(this.operatorModeStateService.state.codePath!);
+  }
+
+  get realms() {
+    if (!this.realmURL) {
+      return [];
+    }
+    return [this.realmURL.href];
```

Comment:
Why do we only want to search in the current realm for a spec?

Reply:
Ahh good catch. I think this should be searching for all available realms that the user has access to that references this code.  https://github.com/cardstack/boxel/pull/1993#discussion_r1900223901 this wud then make more sense to enumerate the different code specs in a radio list too